### PR TITLE
Qbri123 14 add product to cart test

### PR DESCRIPTION
**Pull request by KhraschukZahar**

### CODE REVIEW

File: src/test/java/pages/ShoppingCartPage.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String PRICE_SHOES_FIELD = "//div[contains(@class, 'product-price product-price-js')]//span[contains(@class,'product-price-amount-js')]";
+
+    protected final String SIZE_SHOES_FIELD = "//option[@value = '2.5']";
+
+    protected final String NAME_SHOES_FIELD_CARD = "//h3[@class='item-detail-info-name item-name-js']";
+
+    protected final String PRICE_SHOES_FIELD_CARD = "//div[@class='item-detail-total item-detail-total-js small-6 medium-3 large-1 columns ']";
+
+    protected final String SIZE_SHOES_FIELD_CARD = "//dd[contains(text(),'2.5')]";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:37]: resolved

File: src/test/java/tests/AddToCartTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import base.AbstractBaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pages.HomePage;
+import pages.ShoppingCartPage;
+
+import java.util.List;
+
+public class AddToCartTest extends AbstractBaseTest {
+    List<String> current_price;
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/AddToCartTest.java:12]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String PROCEED_TO_CHECKOUT = "(//div[contains(@class, 'mini-cart-actions cart-actions')])[1]";
+
+    protected final String NAME_SHOES_FIELD = "//h1[@class='product-info-js']";
+
+    protected final String PRICE_SHOES_FIELD = "//div[contains(@class, 'product-price product-price-js')]//span[contains(@class,'product-price-amount-js')]";
+
+    protected final String SIZE_SHOES_FIELD = "//option[@value = '2.5']";
+
+    protected final String NAME_SHOES_FIELD_CARD = "//h3[@class='item-detail-info-name item-name-js']";
```
Comment (Goldbe54) [inline src/test/java/pages/ShoppingCartPage.java:33]: You create a Class for the side window "ShoppingCartPage". Create one more class for "BaseCartPage", and put NAME_SHOES_FIELD_CARD in (+ PRICE_SHOES_FIELD_CARD)

File: src/test/java/pages/ShoppingCartPage.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String ADD_TO_CARD = "//button[@data-add-to-cart-text='add to cart']";
+
+    protected final String PROCEED_TO_CHECKOUT = "(//div[contains(@class, 'mini-cart-actions cart-actions')])[1]";
+
+    protected final String NAME_SHOES_FIELD = "//h1[@class='product-info-js']";
+
+    protected final String PRICE_SHOES_FIELD = "//div[contains(@class, 'product-price product-price-js')]//span[contains(@class,'product-price-amount-js')]";
+
+    protected final String SIZE_SHOES_FIELD = "//option[@value = '2.5']";
```
Comment (Goldbe54) [inline src/test/java/pages/ShoppingCartPage.java:31]: SIZE_SHOES_FIELD must be in the new class "ProductPage"

File: src/test/java/pages/ShoppingCartPage.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String NAME_SHOES_FIELD_CARD = "//h3[@class='item-detail-info-name item-name-js']";
+
+    protected final String PRICE_SHOES_FIELD_CARD = "//div[@class='item-detail-total item-detail-total-js small-6 medium-3 large-1 columns ']";
+
+    protected final String SIZE_SHOES_FIELD_CARD = "//dd[contains(text(),'2.5')]";
+
+    public ShoppingCartPage clickShopNowButton() {
```
Comment (Goldbe54) [inline src/test/java/pages/ShoppingCartPage.java:39]: Rewrite all methods by mehtods in **Main**

File: src/test/java/tests/AddToCartTest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        shoppingCartPage
+                .clickShopNowButton()
+                .clickOnShoes()
+                .clickOnSelectSizeDropdown()
+                .clickOnSelectSize()
+                .clickOnAddToCard();
+
+        current_price = shoppingCartPage.getInfoAboutProductOnHomePage();
+
+        shoppingCartPage.clickOnProceedToCheckout();
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/tests/AddToCartTest.java:28]: resolved

File: src/test/java/base/BasePage.java [hunk 7]
```diff
@@ -16,10 +16,12 @@ public class BasePage extends Assert {
     protected WebDriver driver;
     protected WebDriverWait wait;
     protected Actions actions;
+    protected JavascriptExecutor jse;
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/base/BasePage.java:19]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShoppingCartPage extends BasePage {
+    public ShoppingCartPage(WebDriver driver) {
+        super(driver);
+    }
+
+    protected final String SHOP_NOW_BUTTON = "//a[@href='https://www.vans.co.uk/shop/en-gb/vans-gb/knu-skool#banner=WK23.TheKnuStack.CTA_ShopNow.HP_Hero']";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:15]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String SHOES_PICK = "(//figure[@class='product-block-figure'])[1]";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:17]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String DROPDOWN_SELECT_SIZE = "//div[@class='product-content-form-size-step-dropdown-container custom-dropdown-container']";
+
+    protected final String SELECT_EXACT_SIZE = "//div[@value='2.5']";
+
+    protected final String ADD_TO_CARD = "//button[@data-add-to-cart-text='add to cart']";
+
+    protected final String PROCEED_TO_CHECKOUT = "(//div[contains(@class, 'mini-cart-actions cart-actions')])[1]";
+
+    protected final String NAME_SHOES_FIELD = "//h1[@class='product-info-js']";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:27]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+    protected final String PRICE_SHOES_FIELD = "//div[contains(@class, 'product-price product-price-js')]//span[contains(@class,'product-price-amount-js')]";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:29]: resolved

File: src/test/java/pages/ShoppingCartPage.java (hunk 3 above)
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:33]: resolved

File: src/test/java/pages/ShoppingCartPage.java (hunk 4 above)
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:31]: resolved

File: src/test/java/pages/ShoppingCartPage.java (hunk 5 above)
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java:39]: resolved

**Review by Goldbe54 — COMMENTED** [review]
Use locators and methods of these locators where they are

File: src/test/java/pages/NamePage.java [hunk 12]
```diff
@@ -7,11 +7,11 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.lang.Integer.parseInt;
 import static utils.Utils.getElementText;
 
 public class NamePage extends BasePage {
     protected final String PRODUCT_TITLES_SPAN = "//span[@class ='product-block-name-wrapper']";
+    protected static final String SHOES_PICK = "(//figure[@class='product-block-figure'])[1]";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/NamePage.java:15]: resolved

File: src/test/java/pages/HomePage.java [hunk 13]
```diff
@@ -24,4 +21,5 @@ public void fillBaseSearchField(String searchField) {
     public void clickAllowAllCookie() {
         waitUntilElementToBeClickable(ALLOW_ALL_COOKIE_BUTTON).click();
     }
+
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/HomePage.java:28]: resolved

File: src/test/java/pages/HomePage.java [hunk 14]
```diff
@@ -3,13 +3,10 @@
...
 import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.WebElement;
-
-import java.util.ArrayList;
-import java.util.List;
 
 public class HomePage extends BasePage {
 
+    protected static final String SHOP_NOW_BUTTON = "//a[@href='https://www.vans.co.uk/shop/en-gb/vans-gb/knu-skool#banner=WK23.TheKnuStack.CTA_ShopNow.HP_Hero']";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/HomePage.java:9]: resolved

File: src/test/java/pages/ShoppingCartPage.java [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+        infoAboutShoesOnCart.add(shoesNameOnCart.getText());
+
+        WebElement shoesPriceOnCart = waitUntilVisibilityOfElement(PRICE_SHOES_FIELD_CART);
+        infoAboutShoesOnCart.add(shoesPriceOnCart.getText());
+
+        WebElement shoesSizeOnCart = waitUntilVisibilityOfElement(SIZE_SHOES_FIELD_CART);
+        infoAboutShoesOnCart.add(shoesSizeOnCart.getText());
+
+        return infoAboutShoesOnCart;
+    }
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ShoppingCartPage.java]: resolved

File: src/test/java/base/BaseCartPage.java [hunk 16]
```diff
@@ -0,0 +1,7 @@
+package base;
+
+public class BaseCartPage {
+    public static final String NAME_SHOES_FIELD_CART = "//h3[@class='item-detail-info-name item-name-js']";
+    public static final String PRICE_SHOES_FIELD_CART = "//div[@class='item-detail-total item-detail-total-js small-6 medium-3 large-1 columns ']";
+    public static final String SIZE_SHOES_FIELD_CART = "//dd[contains(text(),'2.5')]";
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/base/BaseCartPage.java]: resolved my friendo!

**Review by Goldbe54 — APPROVED** [review]
Ok

**Review by ArtemKostiuktest — APPROVED** [review]

File: src/test/java/pages/ProductPage.java [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+        WebElement shoesPrice = waitUntilVisibilityOfElement(PRODUCT_PRICE);
+        infoAboutShoes.add(shoesPrice.getText());
+
+        WebElement shoesSize = waitUntilVisibilityOfElement(SIZE_SHOES_FIELD);
+        infoAboutShoes.add(shoesSize.getText());
+
+        return infoAboutShoes;
+
+    }
+
```
[the remark this answers is not in the document]
Comment (KhraschukZahar) [reply src/test/java/pages/ProductPage.java:79]: resolved

**Pull request merged**